### ISSUE TITLE
how can I do auth ?

### ISSUE THREAD

**Issue by medyagh** [docker-library/cassandra#106]
is there a way to create a testuser and I can authenticate with it?

**Comment by demestav** [comment]
In `/etc/cassandra/cassandra.yaml` the `authenticator` is set to `AllowAllAuthenticator` essentially disabling authentication. This needs to be set to `PasswordAuthenticator` before starting (or restarting) Cassandra. When authentication is enabled, you then can login as the default `cassandra` user and create the `testuser`.

Of course it would be better if authentication could be enabled through an environment variable.

**Comment by dag4202** [comment]
This is also cumbersome to configure in a local development, due to #58.

**Comment by medyagh** [comment]
agree with @dag4202 it is very painful !

**Comment by JMurph2015** [comment]
I agree here.  Since Cassandra doesn't have a good way to auto-configure auth, this is incredibly tedious with docker containers right now.

**Comment by evanstucker-hates-2fa** [comment]
I need to be able to configure authentication. Can we place merge one of the various pull requests that fixes this?

Security should be the default, not the special case. Docker images aren't just for developing on laptops anymore - this image is being used in production environments.

**Comment by tianon** [comment]
I'm sorry, but honestly if we're going to do anything more with environment variables it would likely be in the opposite direction (which we won't do due to the backwards compatibility implications) -- modifying a YAML file via shell script is very fragile, and setting complex configuration in this way is not great.  For custom configuration, the only/best recommendation is to provide an appropriate `cassandra.yaml` file to your running instance (via a new image with `FROM` / `COPY` in a short `Dockerfile`, via [Docker Configs](https://docs.docker.com/engine/swarm/configs/), via bind mount, etc).

Closing in favor of https://github.com/docker-library/cassandra/pull/160 + https://github.com/docker-library/docs/pull/1317.
